extensions/git/src/model.ts

```diff
@@ -272,7 +273,14 @@ export class Model implements IRemoteSourceProviderRegistry, IPushErrorHandlerRe
 			}
 
 			const dotGit = await this.git.getRepositoryDotGit(repositoryRoot);
-			const repository = new Repository(this.git.open(repositoryRoot, dotGit), this, this, this.globalState, this.outputChannel);
+			const repository = new Repository(
+				this.git.open(repositoryRoot, dotGit),
+				this,
+				this,
+				this.globalState,
+				this.outputChannel,
+				this.getRepositoryForSubmodule(repositoryRootUri)
```

Comment:
Should we _link_ parent/child repositories in here instead? https://github.com/microsoft/vscode/blob/70953ac5de173c23f58006d57d55ea41ebb782a1/extensions/git/src/model.ts#L334